service_auth/views.py

```diff
@@ -83,12 +88,20 @@ def get(self, request, format=None):
         service.save()
 
         # create new codecov access token
-        create_new_codecov_access_token(user)
+        try:
+            create_new_codecov_access_token(user)
+        except Exception as e:
+            notify_user_of_error(user, channel_id)
```

Comment:
the Response object that may be returned here is not being used? (similarly for the success call)

Reply:
it is being used, it gets returned in the slack channel, this response here is returned in the landing page

Reply:
I mean in the function it does a `return Response({"detail": "..."}, status=404)`, but wouldn't you want to capture that here, something like
```
response = notify_user_of_error(user, channel_id)
if response:
    return response
```

Reply:
Ah so you’re suggesting we move the response to the notify function? imo since that function is specifically designed for Slack notifications, it makes sense to keep its functionality isolated, we would also be passing less params. This approach will help maintain clear separation of concerns

Reply:
Yeah you're right. All I'm saying is if you look at `Line 158 of service_auth/helpers.py` it returns an object, however that is never assigned to a value when you call it at `Line 94 or service_auth/views.py`.

Reply:
Makes sense!